core/consensus/beefy/impl/beefy_impl.cpp

```diff
@@ -424,6 +469,14 @@ namespace kagome::network {
     if (grandpa_finalized.number < *beefy_genesis_) {
       return outcome::success();
     }
+
+    if (grandpa_finalized.number) {
+      if (!fetching_headers_.has_value()) {
```

Comment:
Move to begin of `while (next_digest_ <= grandpa_finalized.number)` loop below, before `findValidators`
```suggestion
    if (not fetching_headers_) {
      if (auto r = block_tree_-> getBlockHash(grandpa_finalized.number); not r or not r.value()) {
```